hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestRatisUnderReplicationHandler.java

```diff
@@ -247,6 +252,131 @@ public void testInsufficientTargetsFoundBecauseOfPlacementPolicy() {
     Assert.assertEquals(1, metrics.getPartialReplicationTotal());
   }
 
+  @Test
+  public void testNoTargetsFoundBecauseOfPlacementPolicyRemoveNone() {
+    policy = ReplicationTestUtil.getNoNodesTestPlacementPolicy(nodeManager,
+        conf);
+    RatisUnderReplicationHandler handler =
+        new RatisUnderReplicationHandler(policy, conf, replicationManager);
+
+    Set<ContainerReplica> replicas
+        = createReplicas(container.containerID(), State.CLOSED, 0);
+
+    ContainerReplica shouldDelete = createContainerReplica(
+        container.containerID(), 0, IN_SERVICE, State.UNHEALTHY);
+    replicas.add(shouldDelete);
+
+    Assert.assertThrows(IOException.class,
+        () -> handler.processAndSendCommands(replicas,
+            Collections.emptyList(), getUnderReplicatedHealthResult(), 2));
+    // No commands send, as there are only 2 replicas available.
+    Assert.assertEquals(0, commandsSent.size());
```

Comment:
Shouldn't the under replication handler make a copy of the closed one in this case, or does it think that there are only two nodes in the cluster total because of the test setup?

Reply:
The placement policy is mocked to always fail to find a new node to push the code into the `removeUnhealthyReplicaIfPossible` method. So this test is there to prove we don't go from 2 replicas to only 1 replica, as that would cause a higher risk of dataloss.